File: packages/sdk/typescript/examples/src/list-api-keys.ts
/**
 * Minimal example: list the API keys for an organization via the SDK.
 *
 * Tokens are intentionally omitted from the list response (the `token` field
 * is only included in the creation response — see `client.apiKeys.create`).
 *
 * Prereq: the SDK must be built (`dist/` populated). From the repo root:
 *   pnpm --filter @latitude-data/sdk build
 *
 * From this directory:
 *   pnpm apiKeys:list
 *
 * From the repo root (workspace filter, how you'd run it under Turbo):
 *   pnpm --filter @examples/sdk-typescript apiKeys:list
 */
import { LatitudeClient } from "@latitude-data/sdk"
import { optionalEnv, requireEnv } from "./env.ts"

const client = new LatitudeClient({
  baseUrl: optionalEnv("LATITUDE_API_BASE_URL"),
  apiKey: requireEnv("LATITUDE_API_KEY"),
})

const { apiKeys } = await client.apiKeys.list()

console.log(`Found ${apiKeys.length} API key(s):`)
console.log(JSON.stringify(apiKeys, null, 2))

File: packages/sdk/python/examples/list_api_keys.py
# Minimal example: list the API keys for an organization via the SDK.
#
# Tokens are intentionally masked in the list response (the full `token`
# field is only included in the creation and detail responses — see
# `client.api_keys.create` / `client.api_keys.get`).
#
# From `packages/sdk/python` (after `uv sync`):
#   uv run --env-file examples/.env python examples/list_api_keys.py

from _env import optional_env, require_env

from latitude_sdk import LatitudeClient

client = LatitudeClient(
    base_url=optional_env("LATITUDE_API_BASE_URL"),
    api_key=require_env("LATITUDE_API_KEY"),
)

result = client.api_keys.list()

print(f"Found {len(result.api_keys)} API key(s):")
print(result.model_dump_json(indent=2))
